docset/winserver2025-ps/smbshare/Set-SmbServerAlternativePort.md

```diff
@@ -0,0 +1,248 @@
+---
+description: Use this topic to help manage Windows and Windows Server technologies with Windows PowerShell.
+external help file: SmbServerAlternativePort.cdxml-help.xml
+Module Name: SmbShare
+ms.date: 02/22/2024
+online version: https://learn.microsoft.com/powershell/module/smbshare/set-smbserveralternativeport?view=windowsserver2025-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Set-SmbServerAlternativePort
+---
+
+# Set-SmbServerAlternativePort
+
+## SYNOPSIS
+Configures the alternative ports for the Server Message Block (SMB) protocol on the local server.
+
+## SYNTAX
+
+```
+Set-SmbServerAlternativePort [-TransportType] <TransportType> [-Port] <UInt16>
+ [-EnableInstances] <Instance> [[-DisableInstances] <Instance>] [-Force]
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Set-SmbServerAlternativePort` cmdlet can be used to set the alternate ports that SMB will use
+for incoming connections that accepts a list of port numbers and updates the SMB configuration
+accordingly.
+
+## EXAMPLES
+
+### Example 1: Configure a new SMB server alternative port
+
+```powershell
+PS C:\\> Set-SmbServerAlternativePort -TransportType QUIC -Port 1
+```
+
+Configures an SMB server alternative port that uses the QUIC transport protocol on port number
+**1**.
+
+## PARAMETERS
+
+### -AsJob
+
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
+complete.
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227967)
+or [Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
+current session on the local computer.
+
+```yaml
+Type: CimSession[]
+Parameter Sets: (All)
+Aliases: Session
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DisableInstances
+
+Disables the creation of multiple instances of the SMB server service and sets the port number
+specified by the **Port** parameter as the default port for the SMB server service.
+
+```yaml
+Type: Instance
+Parameter Sets: (All)
+Aliases:
+Accepted values: None, Default
+
+Required: False
+Position: 4
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -EnableInstances
+
+Enables the creation of multiple instances of the SMB server service, each with its own unique port
```

Comment:
Enables alternative ports on SMB instances. Currently only the Default instance is allowed, so this setting is not used.